dvc/remote/gdrive.py

```diff
@@ -195,10 +207,28 @@ def drive(self):
 
             # Pass non existent settings path to force DEFAULT_SETTINGS loading
             gauth = GoogleAuth(settings_file="")
-            gauth.CommandLineAuth()
 
-            if os.getenv(RemoteGDrive.GDRIVE_USER_CREDENTIALS_DATA):
-                os.remove(self.gdrive_user_credentials_path)
+            try:
+                gauth.CommandLineAuth()
+            except RefreshError as e:
+                raise GDriveAccessTokenRefreshError(
+                    msg="Google Drive's access token refreshment is failed",
+                    cause=e,
+                )
+            except KeyError as e:
+                raise GDriveMissedCredentialKeyError(
+                    msg="Google Drive's user credentials file '{}' "
+                    "misses value for key '{}'".format(
+                        self.gdrive_user_credentials_path, str(e)
+                    )
+                )
+            except (AuthenticationError, Exception) as e:
```

Comment:
do we need to specify both exception types here?

Reply:
I would left `AuthenticationError` here to show that we're caring about such kind of errors.
General `Exception` is handled with the same `msg` since the `msg` suits both contexts.

Do you think it's better to left handling only for general `Exception`?

Reply:
I don't see too much value in this, it's clear that AuthenticationError is handled with the same message if `except Exception` is specified. I wonder if we actually need to catch Exception here at all, you sure that we can get something unexpected _but_ meaningful?

Reply:
I've put `Exception` here initially, exactly, because I've checked inner implementation of PyDrive.auth.GoogleAuth.CommandLineAuth and scared about unpredictable range of different kind of Exceptions, which looks like will be misleading for DVC's users.

[PyDrive.auth.GoogleAuth.CommandLineAuth](https://github.com/gsuitedevs/PyDrive/blob/master/pydrive/auth.py#L244)

It calls [CheckAuth](https://github.com/gsuitedevs/PyDrive/blob/master/pydrive/auth.py#L103) as its decorator ( there different keys are obtained from map without checking for their existence. looks to me danger )

CheckAuth calls a lot thing inside and [LoadCredentials](https://github.com/gsuitedevs/PyDrive/blob/cf58f81efa67cb30f7c9b3da7d304f214a9b6d3b/pydrive/auth.py#L277) can throw `InvalidConfigError`, but so far I was not able to find the way how I can raise it on behalf of dvc user.

I guess, it should be possible to discover much more possible sources of unexpected exceptions.

Reply:
Thanks, @MaxRis . Sounds reasonable to me. I would probably keep just Exception then. If it makes sense to mention that we catch AuthenticationError let's put a comment? I just don't like redundant code that makes me think twice (like is there a _special_ reason behind this, may be I don't understand python enough :) )

Reply:
>  I would probably keep just Exception then. If it makes sense to mention that we catch AuthenticationError let's put a comment? 

Done, thanks!

Reply:
Thanks!